docs/extensions/backupentry.md

```diff
@@ -0,0 +1,43 @@
+
+Before introducing the `BackupEntry` extension resource Gardener was using Terraform in order to create and manage these provider-specific resources (e.g., see [here](https://github.com/gardener/gardener/tree/0.27.0/charts/seed-terraformer/charts/aws-backup)).
+Now, Gardener commissions an external, provider-specific controller to take over this task. You can also refer to backupInfra proposal documentation to get idea about how the transition was done and understand the resource in broader scope.
+
+## What is the lifespan of `BackupEntry`?
+The bucket associated with `BackupEntry` will be created at using `BackupBucket` resource. The `BackupEntry` resource will be created as a part of `Shoot` creation. But resource continue to exists post deletion of `Shoot`. The  `deletionGracePeriod` of `BackupEntry` resource i.e. time after the shoot deletion
+is configurable globally for Gardener via gardener-controller-manager component config. You can find the configuration option in reference.
+
+## What needs to be implemented to support a new infrastructure provider?
+
+As part of the shoot flow Gardener will create a special CRD in the seed cluster that needs to be reconciled by an extension controller, for example:
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: BackupEntry
+metadata:
+  name: shoot--foo--bar
+spec:
+  type: azure
+  region: eu-west-1
+  bucketName: foo
+  secretRef:
+    name: backupprovider
+    namespace: shoot--foo--bar
+```
+
+The `.spec.secretRef` contains a reference to the provider secret pointing to the account that shall be used to create the needed resources. This provider secret will be propagated from `BackupBucket` resource by Shoot controller.
+
+Your controller is supposed to create the `etcd-backup` secret in control-plane namespace for shoot. This secret is supposed to be used by Gardener or eventually etcd-backup-restore component to backup the etcd. The controller implementation should cleanup the objects created under shoot specific prefix in bucket equivalent to name of `BackupEntry` resource.
+
+In order to support a new infrastructure provider you need to write a controller that watches all `BackupBucket`s with `.spec.type=<my-provider-name>`. You can take a look at the below referenced example implementation for the Azure provider.
+
+## References and additional resources
+
+* [`BackupEntry` API Reference](https://gardener.cloud/api-reference/extensions/#extensions.gardener.cloud/v1alpha1.BackupBucket)
+* [Exemplary implementation for the Azure provider](https://github.com/gardener/gardener-extensions/tree/master/controllers/provider-azure/pkg/controller/backupentry)
+* [`BackupBucket` resource documentation](./backupbucket.md)
+* [Shared bucket proposal](../proposals/02-backupinfra.md)
+* [Gardener-controller-manager-component-config API specification](https://github.com/gardener/gardener/blob/master/pkg/controllermanager/apis/config/types.go)
```

Comment:
maybe also provide the permalink for the backup configuration (referring to the related lines in the code).

Reply:
Done